Sources/PackageLoading/ManifestJSONParser.swift

```diff
@@ -315,213 +304,70 @@ enum ManifestJSONParser {
-            // Parse the version and validate that it can be used in the current
-            // manifest version.
-            let versionString: String = try platformJSON.get("version")
-
-            // Get the platform name.
-            let platformName: String = try platformJSON.getJSON("platform").get("name")
-
-            let versionComponents = versionString.split(
-                separator: ".",
-                omittingEmptySubsequences: false
-            )
-            var version: [String.SubSequence] = []
-            var options: [String.SubSequence] = []
-
-            for (idx, component) in versionComponents.enumerated() {
-                if idx < 2 {
-                    version.append(component)
-                    continue
-                }
-
-                if idx == 2, UInt(component) != nil {
-                    version.append(component)
-                    continue
-                }
-
-                options.append(component)
-            }
-
-            let description = PlatformDescription(
-                name: platformName,
-                version: version.joined(separator: "."),
-                options: options.map{ String($0) }
-            )
-
-            // Check for duplicates.
-            if platforms.map({ $0.platformName }).contains(platformName) {
-                // FIXME: We need to emit the API name and not the internal platform name.
-                throw ManifestParseError.runtimeManifestErrors(["found multiple declaration for the platform: \(platformName)"])
-            }
-
-            platforms.append(description)
-        }
-
-        return platforms
-    }
-
     private static func parseTarget(
-        json: JSON,
+        target: Serialization.Target,
         identityResolver: IdentityResolver
     ) throws -> TargetDescription {
-        let providers = try? json
-            .getArray("providers")
-            .map(SystemPackageProviderDescription.init(v4:))
-
-        let pluginCapability = try? TargetDescription.PluginCapability(v4: json.getJSON("pluginCapability"))
-
-        let dependencies = try json
-            .getArray("dependencies")
-            .map {
-                try TargetDescription.Dependency.init(
-                    v4: $0,
-                    identityResolver: identityResolver
-                )
-            }
+        let providers = target.providers?.map { SystemPackageProviderDescription($0) }
+        let pluginCapability = target.pluginCapability.map { TargetDescription.PluginCapability($0) }
+        let dependencies = try target.dependencies.map { try TargetDescription.Dependency($0, identityResolver: identityResolver) }
 
-        let sources: [String]? = try? json.get("sources")
-        try sources?.forEach{ _ = try RelativePath(validating: $0) }
+        try target.sources?.forEach{ _ = try RelativePath(validating: $0) }
+        try target.exclude.forEach{ _ = try RelativePath(validating: $0) }
```

Comment:
iiuc this is essentially validation? if so, would be nicer to abstract these into a `validate` func as an extension to target etc

Reply:
Yep, I think we should probably change this to pass down the relative paths to the next layer, but for this PR I am preserving the existing API of `PackageModel`.